api/api.proto

```diff
@@ -3229,6 +3238,73 @@ message SdkNodeInspectRequest {
   string node_id = 1;
 }
 
+// Options for draining volume attachment from a node
+message SdkNodeDrainAttachmentOptions {
+    // Selector is used for selecting volumes whose
+    // attachment needs to be moved from this node.
+    // The selector could be a list of volume label
+    // key value pairs to select a subset of volumes.
+    map<string, string> selector = 1;
+    // Drain only sharedv4 volumes from a node
+    // By default all volumes will be drained.
+    bool only_sharedv4 = 2;
+}
+
+// Options for draining volume replicas from a node
+message SdkNodeDrainVolumeReplicaOptions {
+    // Selector is used for selecting volumes whose
+    // replicas need to be moved from this node to another.
+    // The selector could be a list of volume label
+    // key value pairs to select a subset of volumes.
+    map<string, string> selector = 1;
+}
+
+// Options for draining/detaching cloud drives from a node
+message SdkNodeDrainCloudDriveOptions {
+    // When set the cloud drives will also get deleted.
+    bool delete_drives = 1;
+}
+
+// Options for draining kvdb from a node
+message SdkNodeDrainKvdbOptions {
+}
+
+
+// Options for draining a node in the cluster
+message SdkNodeDrainOptions {
+    // Perform all the node drain related operations.
+    // This is a single switch to perform all the supported
+    // node drain sub operations.
+    bool all = 1;
+    // Selector is a global selector used while performing
+    // a node drain sub operation. Per option selector take
+    // precedence over global selector
+    map<string, string> selector = 2;
+    // When set volume attachments will be drained from this node.
+    SdkNodeDrainAttachmentOptions drain_attachments = 3;
```

Comment:
to actually drain the attachments, we will have to move the application pods from the node as for the non-shared use case, that's the primary way volumes will be attached on a node.

So should we call this `SdkNodeDrainAppOptions` instead?

For sharedv4, we will drain app pods AND also move the volume attachment.

Reply:
done

Reply:
How we drain the attachments is an implementation detail. Shouldn't the API reflect what it should do and in this case move away all the volume attachments from this node?